quixstreams/sources/community/pubsub/components/pubsub_consumer.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import json
+import logging
+import time
+from concurrent.futures import TimeoutError
+from typing import Callable, Optional
+
+try:
+    from google.cloud.pubsub_v1 import SubscriberClient
+    from google.cloud.pubsub_v1.subscriber.client import Client as SClient
+    from google.cloud.pubsub_v1.subscriber.futures import StreamingPullFuture
+    from google.cloud.pubsub_v1.subscriber.message import Message as PubSubMessage
+    from google.oauth2 import service_account
+    from google.pubsub_v1.types import Subscription
+except ImportError as exc:
+    raise ImportError(
+        f"Package {exc.name} is missing: "
+        'run "pip install quixstreams[pubsub]" to use PubSubSource'
+    ) from exc
+
+
+# Include PubSubMessage here so all google package imports happen in 1 location
+__all__ = ("PubSubConsumer", "PubSubMessage")
+
+logger = logging.getLogger(__name__)
+
+
+class PubSubConsumer:
+    def __init__(
+        self,
+        project_id: str,
+        topic_id: str,
+        subscription_id: str,
+        credentials: Optional[str] = None,
+        max_batch_size: int = 100,
+        commit_timeout_secs: int = 30,
+        default_poll_timeout: float = 5.0,
+        create_subscription: bool = False,
+        async_function: Optional[Callable[[PubSubMessage], None]] = None,
+    ):
+        if credentials:
+            self._credentials = service_account.Credentials.from_service_account_info(
+                json.loads(credentials, strict=False),
+                scopes=["https://www.googleapis.com/auth/cloud-platform"],
```

Comment:
Please check if this is the most minimal scope.

Can `"https://www.googleapis.com/auth/pubsub"` work instead?

Reply:
`"https://www.googleapis.com/auth/pubsub"` will work